backend-db/Rhyolite/Backend/TaskWorker.hs

```diff
@@ -104,3 +105,23 @@ withWorker d work child = do
           Right False -> sleep nextStartVar
         go nextStartVar
   bracket (liftIO $ async $ go initialStartVar) (liftIO . cancel) $ \_ -> child wakeup
+
+
+-- | Run multiple workers in parallel on the same task
+withConcurrentWorkers
+  :: forall r a.
+     Int -- ^ The number of concurrent workers
+  -> NominalDiffTime -- ^ Time delay between restarts after task returns False
+  -> (a -> IO Bool) -- ^ The work to do
+  -> (Int -> a) -- ^ extract the nth argument to
+  -> (IO () -> IO r) -- ^ Continuation to run while the task worker is running.
+  -> IO r
+withConcurrentWorkers n0 d work argFn = runContT $ go n0
+  where
+    go :: Int -> ContT r IO (IO ())
+    go n = do
+      wakeup <- ContT $ withWorker d (work $ argFn n)
+      wakeupRest <- if (n > 1)
+        then go (n-1)
+        else return $ return ()
+      return (wakeup *> wakeupRest)
```

Comment:
I'm not really 100% sure this is exactly right.  I think maybe it should be `liftA2 (||)` or something instead of `(*>)`

Reply:
Would this perhaps be simpler with that `ResourceT`?